### PR TITLE
Add 'option-renderer' property to select2 component

### PR DESCRIPTION
**Pull request by guvial**
Add a new property to select2 component to give access to the rendering functions of the wrapped select2 component to customize rendering of options.

For example, if you want to create a color picker:

In the view:
`<aui-select2-single  :option-renderer="colorOptionRenderer" v-model="xxxx.color.name">
       #foreach($color in $colors )
            <aui-select2-option value="$color.name()">$color.background</aui-select2-option>
       #end
</aui-select2-single>`

"ColorOptionRenderer" method in js:

`dialogVue = new Vue({
	    el : '#your-element',
	    data : {
		...
		colorOptionRenderer : (colorCode) => '<div class="holydev-calendar-color-option" style="background:' + colorCode + '"></div>'
	    },
                ...
`

### CODE REVIEW

File: src/components/select2/AuiSelect2Single.vue [hunk 1]
```diff
@@ -15,7 +15,8 @@
       placeholder: String,
       query: Function,
       value: String,
-      width: String
+      width: String,
+      optionRenderer: Function
```
Comment (damienix) [inline src/components/select2/AuiSelect2Single.vue]: I try to keep original select2 API names here to make it more clear so this should be named `formatResult` perhaps.

File: src/components/select2/AuiSelect2Single.vue [hunk 2]
```diff
@@ -42,7 +43,9 @@
         data: () => ({results: this.options}),
         placeholder: this.placeholder,
         width: this.width,
-        dropdownAutoWidth: this.dropdownAutoWidth
+        dropdownAutoWidth: this.dropdownAutoWidth,
+        formatResult: (state) => state.text,
+        formatSelection: (state) => state.text
```
Comment (damienix) [inline src/components/select2/AuiSelect2Single.vue]: Other places refer to these objects as `options` so please rename to `(option) -> ...` to be consistent.

Comment (damienix) [reply src/components/select2/AuiSelect2Single.vue]: Wait. I don't understand one thing. Currently, (if I understand well from select2 docs, haven't tested yet), this is the default behaviour, so does those 2 lines do anything?
And 2nd, why not just use the passed renderer here similar to this
    
    formatSelection: this.formatResult || (option) => option.text

won't this work?

**Comment by damienix** [comment]
Hi, thanks for the PR - in fact I think this might be useful for me as well.

I'm gonna test it tomorrow, please review my comments.

Cheers!

File: src/components/select2/AuiSelect2Single.vue (hunk 1 above)
Comment (guvial) [reply src/components/select2/AuiSelect2Single.vue]: Yes, I agree, I will rename it

File: src/components/select2/AuiSelect2Single.vue (hunk 2 above)
Comment (guvial) [reply src/components/select2/AuiSelect2Single.vue]: Yes, I will try it

**Comment by guvial** [comment]
Hi,
Sorry for the conflicting files, I have created the pull request from my forked repo but it was not up to date with your last changes. I have now updated my forked repo but I must now spend some time to test and adapt my code. I will also take into consideration your comments below. It is the first time I use "forked repo" in githup, sorry for the confusion. ;-)
Cheers !

**Comment by guvial** [comment]
Hi Damian,
I have prepared a cleaner version without conflicts.
I will close this PR and start from a new one.
Cheers,
Guillaume

**Comment by damienix** [comment]
Sure, I'm looking forward to see it :)

**Pull request closed without merging**